hypothesis-python/src/hypothesis/strategies/_internal/utils.py

```diff
@@ -144,3 +144,14 @@ def accept(*args, **kwargs):
         return accept
 
     return decorator
+
+
+def utf8_encodable(terminal: str) -> bool:
+    try:
+        terminal.encode()
+        return True
+    except UnicodeEncodeError:  # pragma: no cover
+        # Very rarely, a "." in some terminal regex will generate a surrogate
+        # character that cannot be encoded as UTF-8.  We apply this filter to
+        # ensure it doesn't happen at runtime, but don't worry about coverage.
+        return False
```

Comment:
We don't need this, because our regex uses fullmatch so there are no `.`s or terminals, and the main body can't generate surrogate characters either.  And we won't need it for the interval-based logic for equivalent reasons.